src/uct/cuda/cuda_ipc/cuda_ipc_ep.c

```diff
@@ -97,6 +94,8 @@ uct_cuda_ipc_post_cuda_async_copy(uct_ep_h tl_ep, uint64_t remote_addr,
         }
     }
 
+    key->dev_num %= UCT_CUDA_IPC_MAX_PEERS; /* round-robin */
+
```

Comment:
why does dev_num( here uuid index) can go beyond UCT_CUDA_IPC_MAX_PEERS ?

Reply:
This is to accommodate the fact that number of GPUs on the system can grow beyond UCT_CUDA_IPC_MAX_PEERS some day.

Reply:
in that case, we may have set highier value for UCT_CUDA_IPC_MAX_PEERS.
It looks like it is already assumed that num of  GPUs on the system is less than UCT_CUDA_IPC_MAX_PEERS.
https://github.com/openucx/ucx/blob/master/src/uct/cuda/cuda_ipc/cuda_ipc_iface.c#L398

And also it seems it is not right it use the same stream  for multiple devices if the index is the same

Reply:
I've gotten rid of that code in this PR, Deva. 
https://github.com/openucx/ucx/pull/4712/commits/c4116b1b4d5feb0b6d8d4bf3f6de22bb57835a78#diff-ddc86d12c1a391580307c75f307b2c45L394
We're trying to remove that restriction where number of devices discovered could increase because CUDA_VISIBLE_DEVICES masking.